[BUG] Including rmm/rmm.h without cmake, make, make install fails

**Describe the bug**
If a project includes `rmm/rmm.h` without doing cmake, make, make install compilation fails with
```
rmm/include/rmm/detail/memory_manager.hpp:37:30: fatal error: rmm/detail/cnmem.h: No such file or directory
```
This does not happen if the include of `cnmem.h` in `memory_manager.hpp:37` is done via `#include "cnmem.h"` instead of `#include "rmm/detail/cnmem.h"`. For projects which have header only dependency to rmm the cmake, make, make install step is not necessary so it would be desriable if this works.

**Steps/Code to reproduce bug**
Compiling

```
#include <rmm/rmm.h>

int main()
{
        return 0;
}
```

with

```
g++ -I$CUDA_HOME/include -Irmm/include rmm_include_bug.cpp
```

reproduces the error

```
In file included from rmm/include/rmm/rmm.hpp:28:0,
                 from rmm/include/rmm/rmm.h:5,
                 from rmm_include_bug.cpp:1:
rmm/include/rmm/detail/memory_manager.hpp:37:10: fatal error: rmm/detail/cnmem.h: No such file or directory
 #include "rmm/detail/cnmem.h"
          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Expected behavior**
Compilation of the above example works. 

**Environment details:**
 - Environment location: Bare-metal
 - Method of RMM install: from source
 - Output of `print_env.sh` attached as [`rmm_print_env.log`](https://github.com/rapidsai/rmm/files/2820760/rmm_print_env.log)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,14 @@
+# RMM 0.6.0 (Date TBD)
+
+## New Features
+
+## Improvements
+
+## Bug Fixes
+
+ - PR #36 Fix non-cmake builds (make cnmem.h include path more flexible)
+
+
 # RMM 0.5.0 (28 Jan 2019)
 
 ## New Features

--- a/include/rmm/detail/memory_manager.hpp
+++ b/include/rmm/detail/memory_manager.hpp
@@ -34,7 +34,7 @@
 extern "C" {
 #include "rmm/rmm_api.h"
 }
-#include "rmm/detail/cnmem.h"
+#include "cnmem.h"
 
 /** ---------------------------------------------------------------------------*
  * @brief Macro wrapper for CNMEM API calls to return appropriate RMM errors.

